### DISCUSSION TITLE
Add slides to the beginning of the stack (cards effect, autoplay & loop mode)

### DISCUSSION THREAD

**Discussion by hrvstr** [nolimits4web/swiper#7456] [Q&A]
I have a swiper configured with the cards effect, loop mode, autoplay, and navigation buttons.

When I switch slides using the drag gesture a new slide is always pushed to the right side of the stack. This way there is always a slide to the right of the stack to be animated in. When using autoplay or navigation buttons this behavior is not present. After all the slides on the right side are used up new slides just jump into view, which doesn't look nice.

[Here is a demo](https://codesandbox.io/p/sandbox/swiper-effect-cards-forked-frz3m4?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clst3ngiq00073b6h8c51o1hx%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clst3ngip00033b6hrbw0v2l3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clst3ngip00043b6h7twvjxtw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clst3ngip00063b6hipirplw2%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B62.93162926701055%252C37.06837073298945%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clst3ngip00033b6hrbw0v2l3%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clst3ngip00023b6hp55muf7s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A100%252C%2522startColumn%2522%253A24%252C%2522endLineNumber%2522%253A100%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clst3ngip00033b6hrbw0v2l3%2522%252C%2522activeTabId%2522%253A%2522clst3ngip00023b6hp55muf7s%2522%257D%252C%2522clst3ngip00063b6hipirplw2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clst3ngip00053b6hp5n2qrhe%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clst3ngip00063b6hipirplw2%2522%252C%2522activeTabId%2522%253A%2522clst3ngip00053b6hp5n2qrhe%2522%257D%252C%2522clst3ngip00043b6h7twvjxtw%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clst3ngip00043b6h7twvjxtw%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). Disable `autoplay` to see the desired behavior. Enable it to see unwanted behavior.

When dragging slides there is always a slide to the right.

https://github.com/nolimits4web/swiper/assets/4610008/74843f46-389c-4fcc-859d-c50ad733fc27

When using autoplay the next slide jumps into view.

https://github.com/nolimits4web/swiper/assets/4610008/f374db6e-a22f-437f-a92a-b3aa5895345d

Is there any workaround or hack to fix this? There is also an issue about this problem here: https://github.com/nolimits4web/swiper/issues/6920

**Comment by EtienneGuillet** [comment]
@hrvstr by any chance did you find a fix?

**Reply by hrvstr** [reply]
No, sadly not. I just removed the effect again from the slider instance.

**Comment by EtienneGuillet** [comment]
Too bad. Thank you for your answer tho.

**Comment by arafatah** [comment]
Any update on this issues?
